Activate queue message behavior for more intuitive message order

oskarth opened the pull request:
Fixes #2702 

See https://github.com/status-im/status-react/pull/2743 for ground work.

### Summary

Intercept received messages and queue them

- Exposes behavior under QUEUE_MESSAGE_ENABLED flag that is turned off in
production.

- Delays messages to allow for reordering of latecoming messages up to 500ms

### Review notes (optional):
Normal review.

### Testing notes (optional):
1. Make a build with QUEUE_MESSAGE_ENABLED enabled and one without
2. Send lots of messages in 1on1 chat from one person, like `1 2 3 4 5 6 7 8 9`.
3a. With flag off you should eventually see ordering being different due to network uncertainty
3b. With flag on this should essentially be impossible to achieve (or way less likely)

status: ready

yenda commented on src/status_im/utils/pre_receiver.cljs:
```diff
@@ -0,0 +1,44 @@
+       (= clock-value)))
+
+;; TODO: Update docstring after add-message fixed
+(defn start!
```
Should we write a tech debt bounty extracting the rolling window async mechanism into async utils namespace ?

oskarth replied:
@yenda We could, I just don't see exactly how it'd be different than this namespace (without earliest-clock-value-seen? fn). What is the tech debt in your opinion?

oskarth replied:
The delay-msg fn looks pretty canonical to me, and the pre-receiver could maybe be more generalized, but it also seems quite tight with specific problem. Do we have another use case than this in mind right now?

oskarth commented on src/status_im/utils/pre_receiver.cljs:
```diff
@@ -0,0 +1,44 @@
+;; See status-im.test.utils.pre-receiver for justification.
+
+;; TODO(oskarth): Doesn't belong here, mocking real add-msg - fix signature
+(defn- add-message [{:keys [id clock-value] :as msg}]
```
Oops, remove this, not used

oskarth commented:
Rebased on develop with other merged PR. One question is if we should put flag earlier in call chain, and then about possible generic fn that @yenda and @janherich brought up (separate issue). Moving to ready for testing.

oskarth committed:
Intercept received messages and queue them

- Exposes behavior under QUEUE_MESSAGE_ENABLED flag that is turned off in
production.

- Delays messages to allow for reordering of latecoming messages up to 500ms

asemiankevich commented:
1. This is build with QUEUE_MESSAGE_ENABLED=1

Branch: status-im:feature/spike-message-order-2-#2702
Android: https://i.diawi.com/952PHz
iOS: https://i.diawi.com/8Upuv8

2. This is build with QUEUE_MESSAGE_ENABLED=0

Branch: status-im:feature/spike-message-order-2-#2702
Android: https://i.diawi.com/8zYYFe
iOS: https://i.diawi.com/jpES9n
